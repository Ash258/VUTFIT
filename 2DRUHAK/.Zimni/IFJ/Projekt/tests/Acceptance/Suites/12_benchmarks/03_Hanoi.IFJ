' Tower of Hanoi for 4 rodes

function hanoi(n as integer, from as string, _to as string, middle as string) as integer
	dim c as integer
	if n = 1 then
		print !"Move disk 1 from row "; from; !" to rod "; _to; !"\n";
		return 0
	end if
	c = hanoi(n - 1, from, middle, _to)
	print !"Move disk"; n; !" from row "; from; !" to rod "; _to; !"\n";
	c = hanoi(n - 1, middle, _to, from)
end function

scope
    dim c as integer
	c = hanoi(4, !"A", !"B", !"C")
end scope
