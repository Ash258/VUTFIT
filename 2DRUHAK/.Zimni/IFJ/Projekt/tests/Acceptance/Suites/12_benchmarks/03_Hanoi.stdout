Move disk 1 from row A to rod C
Move disk 2 from row A to rod B
Move disk 1 from row C to rod B
Move disk 3 from row A to rod C
Move disk 1 from row B to rod A
Move disk 2 from row B to rod C
Move disk 1 from row A to rod C
Move disk 4 from row A to rod B
Move disk 1 from row C to rod B
Move disk 2 from row C to rod A
Move disk 1 from row B to rod A
Move disk 3 from row C to rod B
Move disk 1 from row A to rod C
Move disk 2 from row A to rod B
Move disk 1 from row C to rod B
